demo/app/ios/Runner/Tests/IntegrationTest.swift

```diff
@@ -99,4 +100,95 @@ class IntegrationTest: XCTestCase {
         try vpInteraction.presentCredential(selectedCreds)
     }
 
+        func testAuthFlow() throws {
+          let trace = OtelNewTrace(nil)
+          let kms = LocalkmsNewKMS(kmsStore(), nil)!
+
+         let resolverOpts = DidNewResolverOpts()
+         resolverOpts!.setResolverServerURI("http://localhost:8072/1.0/identifiers")
+         let didResolver = DidNewResolver(resolverOpts, nil)!
+
+          let crypto = kms.getCrypto()
+
+          let didCreator = DidNewCreator(kms, nil)!
+          let userDID = try didCreator.create("ion", opts: nil)
+
+          // Issue VCs in auth flow
+          let requestAuthURI = ProcessInfo.processInfo.environment["INITIATE_ISSUANCE_URLS_AUTH_CODE_FLOW"]
+          print(requestAuthURI)
+          XCTAssertTrue(requestAuthURI != "", "requestAuthURI:" + requestAuthURI!)
+
+          let openID4CIInteractionArgs = Openid4ciNewInteractionArgs(requestAuthURI, crypto, didResolver)
+
+          let ciOpts = Openid4ciNewInteractionOpts()
+          ciOpts!.add(trace!.traceHeader())
+
+          let ciInteraction = Openid4ciNewInteraction(openID4CIInteractionArgs, ciOpts, nil)
+          XCTAssertNotNil(ciInteraction)
+
+          let authCodeGrant = ciInteraction!.authorizationCodeGrantTypeSupported()
+          XCTAssertTrue(authCodeGrant)
+
+          let scopes = ApiStringArray()
+          scopes!.append("openid")!.append("profile")!
+
+          let opts = Openid4ciNewCreateAuthorizationURLOpts()!.setScopes(scopes!)
+          let authorizationLink = ciInteraction!.createAuthorizationURL("oidc4vc_client", redirectURI: "http://127.0.0.1/callback", opts: opts!, error: nil)
+          XCTAssertTrue(authorizationLink  != "", "authorizationLink:" + authorizationLink)
+
+           var redirectURL = ""
+           let r = Redirect()
+           let url = URL(string: authorizationLink)!
+           r.makeRequest(url: url, callback: { (location) in
```

Comment:
Maybe it better to move that logic to some helper function. It will make test code more clear

Reply:
Aah i missed ur comment will do that in follow up pr